src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public Circle(String color, double radius) {
+        super(color);
+        this.radius = radius;
+    }
+
+    @Override
+    public double getArea() {
+        return Math.PI * radius * radius;
+    }
+
+    @Override
+    public void draw() {
+        String randomColor = new ColorSupplier().getRandomColor();
```

Comment:
with the current implementation, you instantiate `class ColorSupplier` each time the method is being invoked.

think about instantiating ColorSupplier once and reusing it multiple times.